spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/ErrorHandling.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.errorhandling
+
+import org.apache.spark.sql.{Column, DataFrame, SparkSession}
+import za.co.absa.spark.commons.errorhandling.implementations.{ErrorMessageSubmitOnColumn, ErrorMessageSubmitWithoutColumn}
+import za.co.absa.spark.commons.errorhandling.types._
+
+trait ErrorHandling {
+  def register(sparkToRegisterTo: SparkSession): Unit = {}
+
+  def putErrorToColumn(errType: ErrType, errCode: ErrCode, errMessage: ErrMsg, additionalInfo: AdditionalInfo = None): ErrorColumn = {
+    val toSubmit = ErrorMessageSubmitWithoutColumn(errType, errCode, errMessage, additionalInfo)
+    putErrorToColumn(toSubmit)
+  }
+
+  def putErrorToColumn(errType: ErrType, errCode: ErrCode, errMessage: ErrMsg, errColName: ErrColName, additionalInfo: AdditionalInfo = None): ErrorColumn = {
+    val toSubmit = ErrorMessageSubmitOnColumn(errType, errCode, errMessage, errColName, additionalInfo)
+    putErrorToColumn(toSubmit)
+  }
+  def putErrorToColumn(errorMessageSubmit: ErrorMessageSubmit): ErrorColumn
```

Comment:
It's a shame that Scala doesn't have (or, does it?) any kind of nice annotation for abstract methods. In some languages, abstract methods are usually put on the beginning of a given class/interface/trait and properly annotated so that it's clear what's available to use and what's needed to override

Reply:
as an ex-Python developer I would say yes but I'm obviously biased, so I think it's okay keep it as it is :)